pynif/tests/test_context.py

```diff
@@ -20,7 +22,48 @@ def setUpClass(cls):
                 nif:endIndex    "{}"^^xsd:nonNegativeInteger ;
                 nif:isString    "{}" .
         """.format(len(cls.example_text), cls.example_text.replace('"', '\\"'))
-    
+        cls.example_turtle_ContextHashBasedString = """
+            @prefix xsd:   <http://www.w3.org/2001/XMLSchema#> .
+            @prefix itsrdf: <http://www.w3.org/2005/11/its/rdf#> .
+            @prefix nif:   <http://persistence.uni-leipzig.org/nlp2rdf/ontologies/nif-core#> .
+                    
+            <http://www.cse.iitb.ac.in/~soumen/doc/CSAW/doc#hash_0_1411_6218664a3a8c7bed58460e329ddc6904_%20%20%20%20Primary%20Navigati>
+                a                       nif:ContextHashBasedString , nif:Context ;
+                nif:beginIndex  "0"^^xsd:nonNegativeInteger ;
+                nif:endIndex    "{}"^^xsd:nonNegativeInteger ;
+                nif:isString    "{}" .
+        """.format(len(cls.example_text), cls.example_text.replace('"', '\\"'))
+        testdir = os.path.dirname(os.path.abspath(__file__))
+        with open(os.path.join(testdir, 'data/example-contextHashBasedString.ttl'), 'r') as f:
+           cls.example_ContextHashBasedString = f.read()
+        """
+        cls.example_ContextHashBasedString = "@prefix itsrdf: <http://www.w3.org/2005/11/its/rdf#> .
+@prefix nif: <http://persistence.uni-leipzig.org/nlp2rdf/ontologies/nif-core#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+<http://freme-project.eu#hash_19_33_158118325b076b079d3969108872d855_Diego%20Maradona%20is%20fr> a nif:ContextHashBasedString,
+        nif:Phrase ;
+    nif:anchorOf "Diego Maradona" ;
+    nif:beginIndex "0"^^xsd:nonNegativeInteger ;
+    nif:endIndex "14"^^xsd:nonNegativeInteger ;
+    nif:referenceContext <http://freme-project.eu#hash_0_33_cf35b7e267d05b7ca8aba0651641050b_Diego%20Maradona%20is%20fr> ;
+    nif:taMsClassRef <http://dbpedia.org/ontology/SoccerManager> ;
+    itsrdf:taAnnotatorsRef <http://freme-project.eu/tools/freme-ner> ;
+    itsrdf:taClassRef <http://dbpedia.org/ontology/Person>,
+        <http://dbpedia.org/ontology/SportsManager>,
+        <http://nerd.eurecom.fr/ontology#Person> ;
+    itsrdf:taConfidence 9.869993e-01 ;
+    itsrdf:taIdentRef <http://dbpedia.org/resource/Diego_Maradona> .
+
+<http://freme-project.eu#hash_0_33_cf35b7e267d05b7ca8aba0651641050b_Diego%20Maradona%20is%20fr> a nif:Context,
+        nif:ContextHashBasedString ;
+    nif:beginIndex "0"^^xsd:nonNegativeInteger ;
+    nif:endIndex "33"^^xsd:nonNegativeInteger ;
+    nif:isString "Diego Maradona is from Argentina." .
+
+"
+"""
```

Comment:
It looks like this big string is no longer needed and can be removed.

Reply:
Yes, you are right! I also went through the code again to check for any other string or unsed text. The only remaining text are the nif:ContextHashBasedString explanations at context.py and phrase.py .
Also added some more assertions on the test and normalized the style according to the previous tests.